docs/api/permissions.rst

```diff
@@ -0,0 +1,188 @@
+.. _api-permissions:
+
+Permissions
+###########
+
+
+Terminology
+===========
+
+.. glossary::
+
+    Object
+        Anything that can be interacted with. Collections, records, buckets, groups
+        are all objects.
+
+    Principal
+        An entity that can be authenticated. Principals can be individual people,
+        applications, services, or any group of such things.
+
+    Group
+        A group associates a name to a list of principals.
+
+    Permission
+    Permissions
+        A permission is an action that can be performed on an object.
```

Comment:
Is a permission the action *itself*, or is it the *right* to perform an action? Consider:
> A permission is the right to perform an action on an object.

This would imply another terminology item for *action*.
> Action
An action is an operation that can be performed on an object. Examples are "read", "write", and "create".